From e39738835e3496ab3b10697356cb7cc1a5281a91 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Sat, 24 Nov 2018 21:37:03 +0900
Subject: [PATCH 1/2] makefile: add DALLOW_EXPERIMENTAL_API option

Some of latest DPDK APIs introduced to replace deprecated APIs are
still experimental. This update is to add `-DALLOW_EXPERIMENTAL_API`
option to CFLAGS to avoid compile errors for the experimental APIs.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 src/mirror/Makefile  | 1 +
 src/nfv/Makefile     | 1 +
 src/primary/Makefile | 1 +
 src/vf/Makefile      | 1 +
 4 files changed, 4 insertions(+)

diff --git a/src/mirror/Makefile b/src/mirror/Makefile
index a759ad2..614a44c 100644
--- a/src/mirror/Makefile
+++ b/src/mirror/Makefile
@@ -19,6 +19,7 @@ SRCS-y += ../vf/common/command_conn.c ../vf/common/command_proc.c
 SRCS-y += ../vf/common/command_dec.c
 SRCS-y += ../vf/common/ringlatencystats.c ../vf/common/string_buffer.c
 
+CFLAGS += -DALLOW_EXPERIMENTAL_API
 CFLAGS += $(WERROR_FLAGS) -O3
 CFLAGS += -I$(SRCDIR)/../shared
 CFLAGS += -I$(SRCDIR)/../vf/common
diff --git a/src/nfv/Makefile b/src/nfv/Makefile
index 4d5eec6..16dfbbb 100644
--- a/src/nfv/Makefile
+++ b/src/nfv/Makefile
@@ -14,6 +14,7 @@ APP = spp_nfv
 # all source are stored in SRCS-y
 SRCS-y := nfv.c ../shared/common.c
 
+CFLAGS += -DALLOW_EXPERIMENTAL_API
 CFLAGS += $(WERROR_FLAGS) -O3
 CFLAGS += -I$(SRCDIR)/../shared
 
diff --git a/src/primary/Makefile b/src/primary/Makefile
index e8a5d8d..bedea8e 100644
--- a/src/primary/Makefile
+++ b/src/primary/Makefile
@@ -23,6 +23,7 @@ SRCS-y := main.c init.c args.c ../shared/common.c
 
 INC := $(wildcard *.h)
 
+CFLAGS += -DALLOW_EXPERIMENTAL_API
 CFLAGS += $(WERROR_FLAGS) -O3
 CFLAGS += -I$(SRCDIR)/../shared
 
diff --git a/src/vf/Makefile b/src/vf/Makefile
index b0a1fdb..f2a167b 100644
--- a/src/vf/Makefile
+++ b/src/vf/Makefile
@@ -18,6 +18,7 @@ SRCS-y += common/command_conn.c common/command_dec.c common/command_proc.c
 SRCS-y += common/spp_proc.c
 SRCS-y += ../shared/common.c
 
+CFLAGS += -DALLOW_EXPERIMENTAL_API
 CFLAGS += $(WERROR_FLAGS) -O3
 CFLAGS += -I$(SRCDIR)/../shared
 CFLAGS += -I$(SRCDIR)/common
-- 
2.7.4

